src/inverseRobot.js

```diff
@@ -19,12 +19,24 @@
  * inverseRobot(kolli) === { name: 'Kolli', chipVer: '123', wheels: '3' }
  *
  *
- * @param {object} robot
+ * @param {object} robot the robot to be inversed.
  *
- * @return {object}
+ * @return {object} inversed robot.
  */
 function inverseRobot(robot) {
-  // write code here
+  const inversed = {};
+  const values = Object.values(robot);
+
+  // Check for duplicate values
+  if (new Set(values).size !== values.length) {
+    return null;
+  }
+
+  for (const key in robot) {
+    inversed[robot[key]] = key;
+  }
```

Comment:
In this 'for...in...' loop you can just check if the new robot has the property of the old robot and return null if it does. Else add a new property.

I appreciated your knowledge of Set data structure but the current task does not require it yet))
Try to make it easier an more understandable using more simple things